applications/app/templates/serviceWorker.scala.js

```diff
@@ -86,70 +87,76 @@ var isSameHost = function (host) {
     }
 })();
 
-var requestForAsset = (function () {
-    var isAssetRequest = (function () {
-        var assetPathRegex = new RegExp('^@Configuration.assets.path');
-        return function (request) {
-            var url = new URL(request.url);
-            @if(play.Play.isDev()) {
-                return assetPathRegex.test(url.pathname);
-            } else {
-                return assetPathRegex.test(url.href);
+var handleDevBlogRequest = function (event) {
+    // update the crossword on every visist the dev blog
+    if (requestAcceptsHTML(event.request)) {
+        isCacheUpdated().then(function (isUpdated) {
+            if (!isUpdated) {
+                updateOfflineCrosswordCache().then(deleteOldCaches);
             }
-        }
-    })();
-
-    return function (event) {
-        var request = event.request;
-
-        if (isAssetRequest(request)) {
-            // Default fetch behaviour
-            // Cache first for all other requests
-            event.respondWith(
-                caches.match(request)
-                    .then(function (response) {
-                        // Workaround Firefox bug which drops cookies
-                        // https://github.com/guardian/frontend/issues/12012
-                        return response || fetch(request, needCredentialsWorkaround(request.url) ? {
-                            credentials: 'include'
-                        } : {});
-                    })
-            );
+        });
+    };
+
+    event.respondWith(
+        fetch(event.request).catch(function () {
+            return caches.match('/offline-crossword');
+        })
+    );
+};
+
+var isRequestForAsset = (function () {
+    var assetPathRegex = new RegExp('^@Configuration.assets.path');
+    return function (request) {
+        var url = new URL(request.url);
+        @if(play.Play.isDev()) {
+            return assetPathRegex.test(url.pathname);
+        } else {
+            return assetPathRegex.test(url.href);
         }
     }
 })();
 
+var handleAssetRequest = function (event) {
+    // Default fetch behaviour
+    // Cache first for all other requests
+    event.respondWith(
+        caches.match(event.request)
+            .then(function (response) {
+                // Workaround Firefox bug which drops cookies
+                // https://github.com/guardian/frontend/issues/12012
+                return response || fetch(event.request, needCredentialsWorkaround(event.request.url) ? {
+                    credentials: 'include'
+                } : {});
+            })
+    );
+};
+
+var isRequestForGenericPage = function (request) {
+    var url = new URL(request.url);
+    return requestAcceptsHTML(request) && isSameHost(url.host);
+}
+
+// handle requests for generic pages
+var handleGenericPageRequest = function (event) {
+    event.respondWith(
+        fetch(event.request).catch(function () {
```

Comment:
it will, but that's ok. you'd get the browser's default offline page anyway (as discussed offline)